aiida_kkr/tools/multi_imps_data_extract.py

```diff
@@ -0,0 +1,453 @@
+    under any key name e.g. key1, key2. The numbers indicate the a couple
+    first-second (0,1) impurities, second-third (1,2) impurities respectively.
+
+    According to the imagnetic impurity order the class collects data related
+    to the impurities. The data includes J_ij interaction, magnetic moment for
+    indivisual impurity, abs(DMI), DMI components, impurity distance 
+    components and so on.
+
+    """
+    
+    def __init__(self, node: Node, imp_orders: dict, ) -> None:
+        """Instantiate
+        node: kkr_combine_imp node.
+        imp_orders: Order of impurities to extract data.
+                For example, for two impurities from 4-impurity
+                calculation, the dict {key1: 1, key2: 2} indicates socond 
+                and third impurities in the impurity list.
+        """
+        
+        from aiida_kkr.workflows import combine_imps_wc
+       
+        self.J_data_uuid: str = None
+        self.calc_imps: int = None
+        self.imp1: int = None
+        self.imp2: int = None
+        self.offset1: int = None
+        self.offset2: int = None
+        self.il1: int = None
+        self.il2: int = None
+        self.J_ij: float = None
+        self.Dx: float = None
+        self.Dy: float = None
+        self.Dz: float = None
+        self.D: float = None
+        self.rx: float = None
+        self.ry: float = None
+        self.rz: float = None
+        self.r: float = None
+        self.mom1: float = None
+        self.mom2: float = None
+        self.tot_mom: float = None
+       
+        # orders in imp pairs
+        self.imp_order1: int = list(imp_orders.values())[0]
+        self.imp_order2: int = list(imp_orders.values())[1]
+        
+        # To set the value for all attributes
+        if issubclass(node.process_class, combine_imps_wc):
+            self.ExtractData(node)
+        else:
+            raise TypeError(f'The node {node.uuid} is not a'
+                            f'combine_imps_wc subclass')
+
+
+        self.ExtractData(node)
+    
+    
+    def __str__(self):
+        print_str = (f'JijData_uuid: {self.J_data_uuid},\n'
+                     f'Total imps as part of calc: {self.calc_imps},\n'
+                     f'Zimp1: {self.imp1}, Zimp2: {self.imp2},\n'
+                     f'offset1: {self.offset1}, offset2: {self.offset2},\n'
+                     f'ilayer1: {self.il1}, ilayer2: {self.il2},\n'
+                     f'J_ij: {self.J_ij},\n'
+                     f'D_ij: {self.D}, Dx: {self.Dx}, Dy: {self.Dy},\n'
+                     f' Dz:{self.Dz},\n'
+                     f'r_ij: {self.r}, rx: {self.rx}, ry: {self.ry},\n'
+                     f'rz: {self.rz},\n'
+                     f'magnetic moment-1: {self.mom1},\n'
+                     f' magnetic moment-2: {self.mom2},\n'
+                     f'total moment: {self.tot_mom}')
+        return print_str 
+
+    def ExtractData(self, node: Node):
```

Comment:
is a certain version of the combine_imps_wc needed? if so, should we check for the version?